src/main/scala/fi/vm/sade/hakurekisteri/integration/koski/KoskiDataHandler.scala

```diff
@@ -603,6 +605,16 @@ class KoskiDataHandler(
         )
       }
 
+      // Tallennetaan mahdollinen 7-8-valmistava vain silloin, kun valmista ja vahvistettua
+      // perusopetuksen suoritusta ei ollut.
+      if (
+        tallennettavatSuoritukset.isEmpty && params.saveSeiskaKasiJaValmistava && isAlaikainen(
+          henkilo
+        )
+      ) {
+        Await.result(updateOppilaitosSeiskaKasiJaValmistava(koskiHenkiloContainer), 5.seconds)
```

Comment:
This _should_ be independent from the rest of the update process, so could be freely run in parallel, but allowing that causes random test errors when the tests await only for the returned Future.

We could return two Futures here just for the sake for being able to await for both of them in the tests, but maybe it's more elegant to do this synchronously and perhaps take a slight perf hit...?